### PR TITLE
Add docs for azure.max-http-requests

### PR DESCRIPTION
**Pull request by mosabua**
## Description

see title 
## Additional context and related issues

Follow up to #22932

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/file-system-azure.md [hunk 1]
```diff
@@ -42,6 +42,10 @@ system support:
   - Maximum number of concurrent write operations. Defaults to 8.
 * - `azure.max-single-upload-size`
   - [Data size](prop-type-data-size) Defaults to `4MB`.
+* - `azure.max-http-requests`
+  - Maximum [integer](prop-type-integer) number of concurrent HTTP requests to
```
[the remark this answers is not in the document]
Comment (wendigo) [reply docs/src/main/sphinx/object-storage/file-system-azure.md:46]: I/O is multiplexed on the cores, adding more threads won't scale ;)

[the remark this answers is not in the document]
Comment (wendigo) [reply docs/src/main/sphinx/object-storage/file-system-azure.md:46]: But that's the detail no one needs to know

**Review by nineinchnick — APPROVED** [review]

File: docs/src/main/sphinx/object-storage/file-system-azure.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-azure.md:46]: it says .. on every node

[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-azure.md:46]: also .. should we document that min=1

[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/object-storage/file-system-azure.md:46]: Adressed now.

**Comment by mosabua** [comment]
Should be good now @anusudarsan @nineinchnick and @wendigo

**Review by ebyhr — APPROVED** [review]

**Review by nineinchnick — APPROVED** [review]

**Review by anusudarsan — APPROVED** [review]

**Pull request merged**